device_service.go

```diff
@@ -0,0 +1,47 @@
+package main
+
+import (
+	"errors"
+	"log"
+)
+
+type Device struct {
+	Id       int     `json:"id"`
+	Name     string  `json:"name"`
+	Interval int     `json:"interval"`
+	Value    float32 `json:"value"`
+}
+
+type deviceDAO interface {
+	Save(device Device) (Device, error)
+}
+
+type deviceService struct {
+	dao deviceDAO
+}
+
+func (s *deviceService) createDevice(device Device) (Device, error) {
```

Comment:
I think that both, `deviceService` and `createDevice` should be public. There is no reason why they should be hidden and they are part of the public API. I know it does not matter now, but let's make it a good practice.

Reply:
What about deviceHTTPHandler. Should it also be public?